spec.md

```diff
@@ -161,9 +161,11 @@ that contains both context and data).
   system might identify the CRM system as the "source", might further qualify the
   event as 'new-customer-added' in the "event-type" relative to its "namespace",
   and then further qualify the subject of the event (the new record) with the content
-  of this field. The subject is a free-form string defined by the publisher.
+  of this property. The subject is a free-form string defined by the publisher. The
+  property is optional, because the subject might already be self-evident from 
```

Comment:
Perhaps: The property might not be present, because the subject...
?

Reply:
It's a bit weird to tap dance around OPTIONAL here. Shall I uppercase or move it back?

Reply:
I think the bigger concern, at least to me, is saying its OPTIONAL twice - once here and once below. Duplicating normative statements isn't good, its opening the possibility of someone updating one of them but not the other - then we're inconsistent.

Reply:
If we have a constraints section, we should also allow rationalization of the constraint in that section, IMO. I'm moving it back.

Reply:
I don't necessary object to that - I'm just trying to avoid two sections that give lots of explanatory text and people not knowing what text goes where - so they end up duplicating thoughts.

Reply:
Actually, on a side note, this is partially why I tend to prefer to not have a "constraints" section because it forces this kind of split. I'd prefer just a set of sentences with the RFC keywords in the appropriate spots and they can live next to the explanatory text.  But I didn't want to drastically change the original format of the spec when we first pulled it in.  But if people are open to the change I can take a pass at a PR.

Reply:
ping @ac360 - any thoughts on this?

Reply:
Providing descriptions without mentioning the constraints will in many cases be hard. @duglin I agree with you that a split of the text is hard to achieve. On the other hand I see some value in a restrictions section that only lists the restrictions without providing additional text. An implementor of the spec who just wants to quickly check the restrictions of an attribute, might benefit from that. Using the same normative statements in the description would be even better than paraphrasing them, because it would help maintaining consistency between description and constaints.